cmd/oci-runtime-tool/generate.go

```diff
@@ -604,6 +614,80 @@ func parseRlimit(rlimit string) (string, uint64, uint64, error) {
 	return parts[0], uint64(hard), uint64(soft), nil
 }
 
+var deviceType = map[string]bool{
+	"b":true, //a block (buffered) special file
+	"c":true, //a character special file
+	"u":true, //a character (unbuffered) special file
+	"p":true, //a FIFO
+}
+
+// addDevice takes the raw string passed with the --device flag, parses it, and add it
+func addDevice(device string, g *generate.Generator) error {
+	dev := rspec.Device{}
+	for _, s := range strings.Split(device, ",") {
+		parts := strings.SplitN(s, "=", 2)
+
+		if len(parts) != 2 {
+			return fmt.Errorf("Incomplete device arguments (%s)", s)
+		}
+
+		name, value := parts[0], parts[1]
+
+		switch name {
+		case "type":
+			if deviceType[value] {
```

Comment:
As you want to add type checks here, what if an invalid type, we should return the error.